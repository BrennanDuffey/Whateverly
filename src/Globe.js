import React, { Component } from 'react';
import Button from './Button.js';
import AnimalCard from './AnimalCard.js';
import CountryCard from './CountryCard.js'
// import worldMap from './worldmap.jpg'


class Globe extends Component {
  constructor(props) {
    super(props);
    this.state = {}
  }
  
  renderSpecies = () => {
    return this.props.endangeredSpecies
    .filter(species => species.name === this.props.searchInput)
    .map(species => {
      return <AnimalCard {...species} key={species.name}/>
    })
  }

  renderCountry = () => {
    return this.props.countries.filter(country => country.name === this.props.searchInput).map(country => {
      return <CountryCard {...country}  key={country.name}/>
    })
  }

  renderCountrySpecies = () => {
    return this.props.endangeredSpecies
    .filter(species => species.locations.includes(this.props.searchInput))
    .map(species => {
      return <AnimalCard {...species} key={species.name}/>
    })
  }


  render() {
    return(
      <main>
        {
          this.renderSpecies(),
          this.renderCountry()
          // this.renderCountrySpecies()
        }
        {
          this.renderCountrySpecies()
        }
      </main>
    )
  }
}




export default Globe;
{/* <picture>
  {/* <img src={worldMap} alt="World Map" /> 
</picture>
<Button /> */}