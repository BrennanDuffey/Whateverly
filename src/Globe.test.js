import React from 'react';
import Globe from './Globe.js';
import { shallow } from 'enzyme';

const mockEndageredSpecies = [
  {
    "name": "Snow Leopard",
    "genus": "Panthera",
    "endangeredStatus": "Vulnerable",
    "locations": ["China", "Mongolia", "India"],
    "population": 6000,
    "threats": ["Poaching","Climate Change"],
    "image": "https://upload.wikimedia.org/wikipedia/commons/d/d7/Snow_Leopard_in_Ladakh%28_Photo_by_Tashi_Lonchay%29.jpg"
  },
  {
    "name": "Jaguar",
    "genus": "Panthera",
    "endangeredStatus": "Near Threatened",
    "locations": ["Amazonian Brazil", "South America", "Central America", "Mexico" ],
    "population": "Unknown",
    "threats": ["Poaching", "Habitat Loss", "Human Intolerance" ],
    "image": "https://images.unsplash.com/photo-1517825738774-7de9363ef735?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80"
  }
];
const mockCountries = [ 
  {
    "name": "China",
    "population": 1418791076,
    "topThreats": ["Poaching", "Climate Change", "Habitat Loss"],
    "conservationGroups": [
      {"name": "Wildlife Conservation Society China", "link": "https://china.wcs.org"},
      {"name": "Save China's Tigers", "link": "https://savechinastigers.org/"}
    ]   
  },
  {
    "name": "Mongolia",
    "population": 3166244,
    "topThreats": ["Poaching", "Climate Change", "Habitat Loss"],
    "conservationGroups": [
      {"name": "Wildlife Conservation Society - Mongolia", "link": "https://mongolia.wcs.org/"},
      {"name": "Wildlife Science and Conservation Center of Mongolia", "link": "http://www.wscc.org.mn/"} 
    ] 
  }
];

const mockFavAnimals = [];
const mockSearchInput = '';
const mockIsFavorite = false;

const mockAddFav = jest.fn();

describe('Globe', () => {
  
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( 
      <Globe  searchInput={mockSearchInput}
              endangeredSpecies={mockEndageredSpecies} 
              countries={mockCountries}
              addFav={mockAddFav} 
              favAnimals={mockFavAnimals}
              isFavorite={mockIsFavorite}/>
    )
  });

  it('Should render correctly with all the information passed', () => {
      expect(wrapper).toMatchSnapshot();
  });
});