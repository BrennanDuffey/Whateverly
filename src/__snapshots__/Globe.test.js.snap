// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Globe Should render correctly with all the information passed 1`] = `
<main>
  <CountryCardContainer
    addFav={[MockFunction]}
    countries={
      Array [
        Object {
          "conservationGroups": Array [
            Object {
              "link": "https://china.wcs.org",
              "name": "Wildlife Conservation Society China",
            },
            Object {
              "link": "https://savechinastigers.org/",
              "name": "Save China's Tigers",
            },
          ],
          "name": "China",
          "population": 1418791076,
          "topThreats": Array [
            "Poaching",
            "Climate Change",
            "Habitat Loss",
          ],
        },
        Object {
          "conservationGroups": Array [
            Object {
              "link": "https://mongolia.wcs.org/",
              "name": "Wildlife Conservation Society - Mongolia",
            },
            Object {
              "link": "http://www.wscc.org.mn/",
              "name": "Wildlife Science and Conservation Center of Mongolia",
            },
          ],
          "name": "Mongolia",
          "population": 3166244,
          "topThreats": Array [
            "Poaching",
            "Climate Change",
            "Habitat Loss",
          ],
        },
      ]
    }
    endangeredSpecies={
      Array [
        Object {
          "endangeredStatus": "Vulnerable",
          "genus": "Panthera",
          "image": "https://upload.wikimedia.org/wikipedia/commons/d/d7/Snow_Leopard_in_Ladakh%28_Photo_by_Tashi_Lonchay%29.jpg",
          "locations": Array [
            "China",
            "Mongolia",
            "India",
          ],
          "name": "Snow Leopard",
          "population": 6000,
          "threats": Array [
            "Poaching",
            "Climate Change",
          ],
        },
        Object {
          "endangeredStatus": "Near Threatened",
          "genus": "Panthera",
          "image": "https://images.unsplash.com/photo-1517825738774-7de9363ef735?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80",
          "locations": Array [
            "Amazonian Brazil",
            "South America",
            "Central America",
            "Mexico",
          ],
          "name": "Jaguar",
          "population": "Unknown",
          "threats": Array [
            "Poaching",
            "Habitat Loss",
            "Human Intolerance",
          ],
        },
      ]
    }
    favAnimals={Array []}
    searchInput=""
  />
  <AnimalCardContainer
    addFav={[MockFunction]}
    endangeredSpecies={
      Array [
        Object {
          "endangeredStatus": "Vulnerable",
          "genus": "Panthera",
          "image": "https://upload.wikimedia.org/wikipedia/commons/d/d7/Snow_Leopard_in_Ladakh%28_Photo_by_Tashi_Lonchay%29.jpg",
          "locations": Array [
            "China",
            "Mongolia",
            "India",
          ],
          "name": "Snow Leopard",
          "population": 6000,
          "threats": Array [
            "Poaching",
            "Climate Change",
          ],
        },
        Object {
          "endangeredStatus": "Near Threatened",
          "genus": "Panthera",
          "image": "https://images.unsplash.com/photo-1517825738774-7de9363ef735?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&w=1000&q=80",
          "locations": Array [
            "Amazonian Brazil",
            "South America",
            "Central America",
            "Mexico",
          ],
          "name": "Jaguar",
          "population": "Unknown",
          "threats": Array [
            "Poaching",
            "Habitat Loss",
            "Human Intolerance",
          ],
        },
      ]
    }
    favAnimals={Array []}
    searchInput=""
  />
</main>
`;
